functions/routes/index.js

```diff
@@ -4,13 +4,20 @@ const router = express.Router();
 const authGuard = require('../util/authGuard');
 
 const {
-    login, signUp, image, getAll,
+    login,
+    signUp,
+    image,
+    getAll,
+    getUserById,
+    AuthUserCredentials,
 } = require('../controllers/user/index');
 
 // user Routes
 router.post('/signup', signUp);
 router.post('/login', login);
 router.put('/user/image', authGuard, image);
 router.get('/users', getAll);
+router.get('/user', authGuard, AuthUserCredentials);
+router.get('/user/:userName', getUserById);
```

Comment:
Okay i see why you added the function above.
But you can specify it by the endpoint name.
you can do something like this  `router.get('/getAuthUser', authGuard, getUserById);`